### PR TITLE
fix: hybird compilation issue on windows

### PR DESCRIPTION
**Pull request by ShyunnY**
[Fix](https://github.com/apache/skywalking/issues/12068)

### CODE REVIEW

File: go.work.sum [hunk 1]
```diff
@@ -389,10 +389,16 @@ contrib.go.opencensus.io/exporter/prometheus v0.4.1 h1:oObVeKo2NxpdF/fIfrPsNj6K0
 dmitri.shuralyov.com/gpu/mtl v0.0.0-20190408044501-666a987793e9 h1:VpgP7xuJadIUuKccphEpTJnWhS2jkQyMt6Y7pJCD7fY=
 dubbo.apache.org/dubbo-go/v3 v3.0.4/go.mod h1:GVCUBZwVkqFJFCtSgIUS7APbv2RBT4KB7sNuLK7D+ko=
 entgo.io/ent v0.9.0 h1:2S1zfpMMW6p+wctj6kcYUprNPNjLWFW06T5MdyAfmWc=
+github.com/99designs/go-keychain v0.0.0-20191008050251-8e49817e8af4 h1:/vQbFIOMbk2FiG/kXiLl8BRyzTWDw7gX/Hz7Dd5eDMs=
```
Comment (mrproliu) [inline go.work.sum:392]: Why do we need to update this file? It looks like not related to this PR. Please restore it.

**Review by mrproliu — COMMENTED** [review]
Thanks! Could you please also update the [CHANGES.md](https://github.com/apache/skywalking-go/blob/main/CHANGES.md). Also, would you help to add a new plugin test in GHA for testing whether the go agent works on the windows or not?

**Comment by wu-sheng** [comment]
Do we have a Windows to verify?

**Comment by ShyunnY** [comment]
> 我们有 Windows 来验证吗？

I will write e2e tests  :)

**Comment by mrproliu** [comment]
Will you write in this PR or a new one? @ShyunnY

**Comment by ShyunnY** [comment]
> Will you write in this PR or a new one? @ShyunnY

Since it may be a while before I can write the e2e tests, I thought I'd start this work in a new PR. 
Thank you for your review!   :)

**Comment by wu-sheng** [comment]
If CI can't be added here, we need a manul test. @mrproliu

**Comment by mrproliu** [comment]
Could you help to add some screenshot in here? @ShyunnY

**Comment by ShyunnY** [comment]
Block

**Comment by ShyunnY** [comment]
cc @wu-sheng 

Hello! When I was fixing [issue](https://github.com/apache/skywalking/issues/12068), I found that go-agent had other problems, which made it impossible to carry out hybrid compilation under the windows platform.
I think this PR is only to fix the problem that go-agent can work normally under the Windows platform, not to solve the problem of [issue](https://github.com/apache/skywalking/issues/12097).

IMO, I think we should merge this PR first, and I will submit a new PR to solve the [issue](https://github.com/apache/skywalking/issues/12097) problem in the future.

**Comment by wu-sheng** [comment]
To be honest, as you are fixing Windows compiling, the new resolved issue should be fixed as well.

And this PR has CI pipeline failing, which can't be merged by anyone, and doesn't fit SkyWalking traditions.

**Comment by wu-sheng** [comment]
You could use one PR to fix two issues, which is fine.
The expectation in SkyWalking is, we prefer PR resolves known relative issues once for all.

**Comment by ShyunnY** [comment]
@wu-sheng 

Sorry, since this is my first time participating in a `skywalking` project, I may not be familiar with the `skywalking` style, but today I understand it.

Thanks for your reply, I will work on this as soon as possible.

**Comment by Superskyyy** [comment]
One can add windows Github action to the .github workflows, just I'm not sure if our e2e pipeline can work on windows.

**Comment by wu-sheng** [comment]
I don't think e2e tool works. But at least, we could have compiling passed on Windows CI.

**Review by mrproliu — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
I can't see Windows platform CI compiling.

**Comment by mrproliu** [comment]
Could you upload some screenshots here? @ShyunnY

**Comment by mrproliu** [comment]
@ShyunnY basic finished the compile phase on windows, but the tracing data not reported for now, he is working on it.

**Comment by ShyunnY** [comment]
Follow the steps of [docs](https://skywalking.apache.org/zh/2023-06-01-quick-start-with-skywalking-go-agent/).  We expect it to execute normally under windows.

actual:
image

Project binaries successfully compile
